src/client.ts

```diff
@@ -0,0 +1,162 @@
+    if (!exec)
+        point = "exec/";
+
+    return podmanCall(con, "libpod/" + point + id + "/resize", "POST", args);
+}
+
+function parseImageInfo(info: JsonObject): JsonObject {
+    const image: JsonObject = {};
+
+    if (info.Config) {
+        const config = info.Config as JsonObject;
+        image.Entrypoint = config.Entrypoint;
+        image.Command = config.Cmd;
+        image.Ports = Object.keys((config.ExposedPorts as JsonObject) || {});
+        image.Env = config.Env || [];
+    }
+    image.Author = info.Author;
+
+    return image;
+}
+
+export function getImages(con: Connection, id?: string) {
+    const options: JsonObject = {};
+    if (id)
+        options.filters = JSON.stringify({ id: [id] });
+    return podmanJson(con, "libpod/images/json", "GET", options)
+            .then(reply => {
+                const images: JsonObject = {};
+                const promises: Promise<JsonObject|JsonValue>[] = [];
+
+                for (const image of reply as JsonObject[]) {
+                    images[image.Id as string] = image;
+                    promises.push(podmanJson(con, "libpod/images/" + image.Id + "/json", "GET", {}));
+                }
+
+                return Promise.all(promises)
+                        .then(replies => {
+                            for (const info of replies as JsonObject[]) {
+                                const imageId = info.Id as string;
+                                const existingImage = images[imageId] as JsonObject || {};
+                                images[imageId] = { uid: con.uid, ...existingImage, ...parseImageInfo(info) };
+                            }
+                            return images;
+                        });
+            });
+}
+
+export function getPods(con: Connection, id?: string) {
+    const options: JsonObject = {};
+    if (id)
+        options.filters = JSON.stringify({ id: [id] });
+    return podmanJson(con, "libpod/pods/json", "GET", options);
+}
+
+export const delImage = (con: Connection, id: string, force: boolean) => podmanJson(con, "libpod/images/" + id, "DELETE", { force });
+
+export const untagImage = (con: Connection, id: string, repo: string, tag: string) => podmanCall(con, "libpod/images/" + id + "/untag", "POST", { repo, tag });
+
+export function pullImage(con: Connection, reference: string) {
+    return new Promise<void>((resolve, reject) => {
+        podmanCall(con, "libpod/images/pull", "POST", { reference })
+                .then(r => {
+                    // Need to check the last response if it contains error
+                    const responses = r.trim().split("\n");
+                    const response = JSON.parse(responses[responses.length - 1]);
+                    if (response.error) {
+                        response.message = response.error;
+                        reject(response);
+                    } else if (response.cause) // present for 400 and 500 errors
+                        reject(response);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2214-e5ac0ddf-20250801-075515-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)